### PR TITLE
Card v2 changes

### PR DESCRIPTION
**Pull request by MI6-451**
## Summary of the changes
Adds subheading prop to add message below heading/title, adds expandable prop to allow an expandable area and control, adds adornment, image-top, image-mid, expanded-content, interaction-button and interaction-controls slots to allow for interactive cards.

BREAKING CHANGE: "icon" slot previously used to add an icon to the top left hand corner of the card has been changed to "card-icon" to stop "icon" slot from icon buttons interfering with the card icon

Edit: The above breaking change has been resolved with a different fix. 

## Related issue
#245 
#213 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by GCHQ-Developer-465 — COMMENTED** [review]
It looks great! Great to see Cards V2

File: packages/web-components/src/components.d.ts [hunk 1]
```diff
@@ -1975,6 +1983,10 @@ declare namespace LocalJSX {
           * If `true`, the card will be disabled if it is clickable.
          */
         "disabled"?: boolean;
+        /**
+          * If `true`, the card will have an expandable area and expansion toggle button.
+         */
+        "expandable"?: boolean;
```
[the remark this answers is not in the document]
Comment (MI6-451) [reply packages/web-components/src/components.d.ts:1989]: Yup, good spot, will get that fixed.

[the remark this answers is not in the document]
Comment (MI6-451) [reply packages/web-components/src/components.d.ts:1989]: Sorry, I only just noticed this in the components.d.ts file which is automatically generated and the duplication/repetition of components and props is expected behaviour.

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
Looks great, just noticed a few small things

**Review by GCHQ-Developer-847 — DISMISSED** [review]

**Comment by jd239** [comment]
Something I have noticed recently is if you wrap the full width variant in a div with display:flex, the component will not stretch 100%.
A ticket has been created for this on select https://github.com/mi6/ic-ui-kit/issues/315. It's worth making the update on Card as well. Think there needs to be 100% width on the host as well when full width is set.

**Review by jd239 — CHANGES_REQUESTED** [review]
Nice update to cards. Just a couple comments.
Also the difference with the Hero Loki tests are surprising. Do you know why that has happened?

File: packages/web-components/src/components/ic-card/ic-card.tsx [hunk 2]
```diff
@@ -188,14 +217,68 @@ export class Card {
...
+            <slot name="interaction-controls"></slot>
+            {expandable && (
+              <ic-button
+                class={{
+                  ["toggle-button"]: true,
+                  ["toggle-button-closed"]: !this.areaExpanded,
+                  ["toggle-button-expanded"]: this.areaExpanded,
+                }}
+                variant="icon"
+                disableTooltip
```
[the remark this answers is not in the document]
Comment (MI6-451) [reply packages/web-components/src/components/ic-card/ic-card.tsx:267]: My reasoning for disabling the tooltip on the 'toggle' icon was that it appears to be what we have done in other similar cases, e.g. Footer grouped links and Side Navigation groups. Happy to make the change if we don't think this is correct in general or in this case.

[the remark this answers is not in the document]
Comment (jd239) [reply packages/web-components/src/components/ic-card/ic-card.tsx:267]: Valid point. Let's leave it how it is.

**Comment by MI6-451** [comment]
> Nice update to cards. Just a couple comments. Also the difference with the Hero Loki tests are surprising. Do you know why that has happened?

The hero loki tests have changed as they have a card in some examples so the changes to cards have appeared there. Looks to just be a small change to padding/margin between the message and the heading which causes the card to be slightly bigger but I believe this is correct.

**Comment by jd239** [comment]
Little concerned about the breaking change as I know the first iteration of cards is being used by dev teams at the moment. I've just tried with using the icon slot for the card in the top left and icon within the button and had no issues. This was for the web component variant. What slot interference were you getting?

File: packages/web-components/src/utils/helpers.ts [hunk 3]
```diff
@@ -419,7 +419,7 @@ export const hasValidationStatus = (
 };
 
 export const isSlotUsed = (element: HTMLElement, slotName: string): boolean => {
-  return !!element.querySelector(`[slot="${slotName}"]`);
+  return Array.from(element.children).some(child => child.getAttribute('slot') === slotName);
```
Comment (jd239) [inline packages/web-components/src/utils/helpers.ts:422]: Might be worth adding a check that the elements.children is HTMLCollection[] so we do not get undefined or null being passed into Array.from()
```javascript
  return HTMLCollection.prototype.isPrototypeOf(element.children)
    ? Array.from(element.children).some(el => el.getAttribute('slot') === slotName)
    : false;
```

File: packages/web-components/src/components/ic-select/readme.md [hunk 4]
```diff
@@ -54,13 +54,6 @@ Type: `Promise<void>`
 
 
 
-## CSS Custom Properties
```
Comment (jd239) [inline packages/web-components/src/components/ic-select/readme.md:57]: The readme is doing that strange thing were it removes items. Just need to run the build one more time.

**Review by jd239 — CHANGES_REQUESTED** [review]
Nearly there. Two small comments

File: packages/web-components/src/utils/helpers.ts (hunk 3 above)
Comment (MI6-451) [reply packages/web-components/src/utils/helpers.ts:422]: As far as I can tell this check shouldn't be necessary as the helper function handles null/undefined by creating an empty array if the element has no children.

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**